step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java

```diff
@@ -192,7 +196,11 @@ public void createOrUpdateCompoundIndex(Map<String, Integer> fields) {
 
 	@Override
 	public void rename(String newName) {
-		// TODO Auto-generated method stub
+		if (parentFactory != null) {

```

Comment:
I assume we should change the value of the field "name" too

Reply:
Correct, fix in next commit